### PR TITLE
Add CORS Support and update Spring framework to 4.2.6

### PR DESCRIPTION
**Pull request by GurubaranVS**

### CODE REVIEW

File: proxyui/src/main/java/com/groupon/odo/containers/HttpProxyContainer.java [hunk 1]
```diff
@@ -26,16 +26,36 @@
...
+import org.springframework.web.filter.CorsFilter;
 
 @Configuration
 public class HttpProxyContainer extends GenericProxyContainer {
+    @Bean
+    public FilterRegistrationBean corsFilter() {
+        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
+        CorsConfiguration config = new CorsConfiguration();
+        config.setAllowCredentials(true);
+        config.addAllowedOrigin("http://localhost:8888");
```
Comment (jungc) [inline proxyui/src/main/java/com/groupon/odo/containers/HttpProxyContainer.java]: Can you double check with richard to see if we want the port number hardcoded?
There was problem related to using specific port number

File: proxyui/src/main/java/com/groupon/odo/controllers/SimpleCORSFilter.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        response.setHeader("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+        chain.doFilter(req, res);
+    }
+    @Override
+    public void init(FilterConfig filterConfig) {}
+    @Override
+    public void destroy() {}
+
+
+}
```
Comment (jungc) [inline proxyui/src/main/java/com/groupon/odo/controllers/SimpleCORSFilter.java]: newline

File: proxyui/src/main/java/com/groupon/odo/controllers/SimpleCORSFilter.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.groupon.odo.controllers;
+import java.io.IOException;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.*;
```
Comment (djrenfro) [inline proxyui/src/main/java/com/groupon/odo/controllers/SimpleCORSFilter.java]: Don't use star imports

File: proxyui/src/main/java/com/groupon/odo/containers/HttpProxyContainer.java (hunk 1 above)
Comment (djrenfro) [reply proxyui/src/main/java/com/groupon/odo/containers/HttpProxyContainer.java]: Should avoid 8888 in general, its a pretty common port for usage, pick a less used one

Comment (GurubaranVS) [reply proxyui/src/main/java/com/groupon/odo/containers/HttpProxyContainer.java]: changed to 8052. Also talked to Richard regarding the hardcoding of the port in the Proxy UI. He mentioned that it is fine by him.

**Pull request merged**